File: src/data/courses/dsa/graphs/graph-fundamentals/build-graph/code/python/build_graph.py
# ============================================================
# Build Graph — Adjacency List
# ============================================================
from collections import defaultdict
class Solution:
    def buildGraph(self, n, edges):
        adj = defaultdict(list)
        for u, v in edges:
            adj[u].append(v)
            adj[v].append(u)  # Undirected
        return adj

File: src/data/courses/dsa/graphs/graph-fundamentals/build-graph/code/java/BuildGraph.java
// ============================================================
// Build Graph — Adjacency List
// ============================================================
import java.util.*;
class Solution {
    public List<List<Integer>> buildGraph(int n, int[][] edges) {
        List<List<Integer>> adj = new ArrayList<>();
        for (int i = 0; i < n; i++) adj.add(new ArrayList<>());
        for (int[] e : edges) {
            adj.get(e[0]).add(e[1]);
            adj.get(e[1]).add(e[0]); // Undirected
        }
        return adj;
    }
}
